tests/testsuite/patch.rs

```diff
@@ -2914,3 +2914,120 @@ Caused by:
+                [dependencies]
+                foo = { path = "vendor/foo" }
+                bar = { path = "vendor/bar" }
+
+                [patch.crates-io]
+                once_cell_0_1 = { package = "once_cell", path = "vendor/once_cell_0_1" }
+                once_cell_1_19 = { package = "once_cell", path = "vendor/once_cell_1_19" }
+            "#,
+        )
+        .file(
+            "src/main.rs",
+            r#"
+                extern crate foo;
+                extern crate bar;
+                fn main() {
+                    foo::print_version();
+                    bar::print_version();
+                }
+            "#,
+        )
+        .file(
+            "vendor/bar/src/lib.rs",
+            r#"
+                extern crate once_cell;
+                pub fn print_version() {
+                    dbg!(once_cell::version());
+                }
+            "#,
+        )
+        .file(
+            "vendor/bar/Cargo.toml",
+            r#"
+                [package]
+                name = "bar"
+                version = "0.1.0"
+                edition = "2015"
+
+                [dependencies]
+                once_cell = "1.19"
+            "#,
+        )
+        .file(
+            "vendor/foo/src/lib.rs",
+            r#"
+                extern crate once_cell;
+                pub fn print_version() {
+                    dbg!(once_cell::version());
+                }
+            "#,
+        )
+        .file(
+            "vendor/foo/Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.1.0"
+                edition = "2015"
+
+                [dependencies]
+                once_cell = "0.1"
+            "#,
+        )
+        .file(
+            "vendor/once_cell_0_1/src/lib.rs",
+            r#"
+                pub fn version() -> &'static str {
+                    "0.1.0"
+                }
+            "#,
+        )
+        .file(
+            "vendor/once_cell_0_1/Cargo.toml",
+            &basic_manifest("once_cell", "0.1.0"),
+        )
+        .file(
+            "vendor/once_cell_1_19/src/lib.rs",
+            r#"
+                pub fn version() -> &'static str {
+                    "1.19.0"
+                }
+            "#,
+        )
+        .file(
+            "vendor/once_cell_1_19/Cargo.toml",
+            &basic_manifest("once_cell", "1.19.0"),
+        )
+        .build();
+
+    p.cargo("check")
+        .with_stderr(
+            "\
+[UPDATING] crates.io index
+[LOCKING] 5 packages to latest compatible versions
+[CHECKING] once_cell [..] ([CWD]/vendor/once_cell_[..])
+[CHECKING] once_cell [..] ([CWD]/vendor/once_cell_[..])
+[CHECKING] [..] v0.1.0 ([CWD]/vendor/[..])
+[CHECKING] [..] v0.1.0 ([CWD]/vendor/[..])
+[CHECKING] cargo-bug-repo v0.1.0 ([CWD])
+[FINISHED] `dev` profile [unoptimized + debuginfo] target(s) in [..]
+",
+        )
+        .run();
+
+    p.cargo("check").with_stderr("[FINISHED] [..]").run();
```

Comment:
The fact that the following commit doesn't change the test, I'm assuming the test is failing within this commit.

Tests should pass in every commit.  Please update this to be passing and then update the test in the following commit to still be passing.  Showing how the test changes helps to test the test and "documents" for people looking at this the exact change in behavior.

Reply:
Fixed :-)